dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/retry/FailedRetryTaskTest.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.dubbo.registry.retry;
+
+import org.apache.dubbo.common.Constants;
+import org.apache.dubbo.common.URL;
+import org.apache.dubbo.common.utils.NetUtils;
+import org.apache.dubbo.common.utils.StringUtils;
+import org.apache.dubbo.registry.NotifyListener;
+import org.apache.dubbo.registry.RegistryService;
+import org.apache.dubbo.registry.support.FailbackRegistry;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.BDDMockito.mock;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class FailedRetryTaskTest {
```

Comment:
Looks like this class is a supplement to FailbackRegistryTest some scenarios? Why not put this code in FailbackRegistryTest?

Reply:
> Looks like this class is a supplement to FailbackRegistryTest some scenarios? Why not put this code in FailbackRegistryTest?

I checked the code again. The functions of this class and some of the functions of FailbackRegistryTest InvokerTest are repetitive. My intention was to test only failed retries, and i will delete this code.